t/Legacy_And_Test2/preload_diag_note.t

```diff
@@ -1,6 +1,11 @@
 use strict;
 use warnings;
 
+if ($] lt "5.008000") {
```

Comment:
I don't think you need string comparison. You can just write `$] < 5.008`

Reply:
It is a good idea to compare perl version via string comparison. Floating-point comparison depends on precision which can be in some cases lost. I was told that due to this fact for `$]` should be always used string version.

Reply:
Now I'm looking it is also written in perl documentation:
https://perldoc.perl.org/perlvar.html#%24%5d
> When comparing `$]` , string comparison operators are **highly recommended**. The inherent limitations of binary floating point representation can sometimes lead to incorrect comparisons for some numbers on some architectures.

Reply:
My recommendation would be to use `"$]" < 5.008000`.  It should really be a numeric comparison, but there are some issues with how `$]` is defined on earlier perl versions which forcing it through stringification will resolve.